core/src/main/resources/reference.conf

```diff
@@ -475,6 +475,58 @@ datastax-java-driver {
       # trace id.
       consistency = ONE
     }
+
+    # A session-wide component that controls the rate at which requests are executed.
+    #
+    # Implementations vary, but throttlers generally track a metric that represents the level of
+    # utilization of the session, and prevent new requests from starting when that metric exceeds a
+    # threshold. Pending requests may be enqueued and retried later.
+    #
+    # From the public API's point of view, this process is mostly transparent: any time that the
+    # request is throttled is included in the session.execute() or session.executeAsync() call.
+    # Similarly, the request timeout encompasses throttling: the timeout starts ticking before the
+    # throttler has started processing the request; a request may time out while it is still in the
+    # throttler's queue, before the driver has even tried to send it to a node.
+    #
+    # The only visible effect is that a request may fail with a RequestThrottlingException, if the
+    # throttler has determined that it can neither allow the request to proceed now, nor enqueue it;
+    # this indicates that your session is overloaded.
+    throttler {
+      # The following implementations are provided out of the box (all located in the package
+      # com.datastax.oss.driver.internal.core.session.throttling):
+      #
+      # - PassThroughRequestThrottler: does not perform any kind of throttling, all requests are
+      #   allowed to proceed immediately. Required options: none.
+      #
+      # - ConcurrencyLimitingRequestThrottler: limits the number of requests that can be executed
+      #   in parallel. Required options: max-concurrent-requests, max-queue-size.
+      #
+      # - RateLimitingRequestThrottler: limits the request rate per second. Required options:
+      #   max-requests-per-second, max-queue-size, drain-interval.
+      class = com.datastax.oss.driver.internal.core.session.throttling.PassThroughRequestThrottler
```

Comment:
Was just thinking that other classes, like `ConstantSpeculativeExecutionPolicy` are in the `api.core` package rather than `internal`.  What should the policy be on this?  Should implementations we offer for use in config files be in the api package or internal?

Another thing that wasn't sure if we discussed or not in the past:  Should we consider having a default package namespace when an instance is called for?  Similar to cassandra.yaml's authenticator/authorizer settings, instead of providing `org.apache.cassandra.auth.AllowAllAuthenticator`, one could simply specify `AllowAllAuthenticator`.

Reply:
Agreed on both points. I've logged [JAVA-1774](https://datastax-oss.atlassian.net/browse/JAVA-1774) and [JAVA-1775](https://datastax-oss.atlassian.net/browse/JAVA-1775).